tests/security/swtpm/swtpm_env_setup.pm

```diff
@@ -26,38 +26,33 @@ sub run {
         select_serial_terminal;
     }
 
-    # Install the required packages for libvirt environment setup
     zypper_call("in qemu libvirt swtpm virt-install virt-manager wget gnutls");
 
-    # Start libvirtd daemon and start the default libvirt network
     assert_script_run("systemctl start libvirtd");
     assert_script_run("virsh net-start default");
     assert_script_run("systemctl is-active libvirtd");
     assert_script_run("virsh net-list | grep default | grep active");
 
-    # Download the pre-installed guest images and sample xml files
+    # Define image paths
     my $image_path = '/var/lib/libvirt/images';
+    my $hdd_swtpm = get_required_var('HDD_2');
     my $legacy_image = 'swtpm_legacy@64bit.qcow2';
     my $uefi_image = 'swtpm_uefi@64bit.qcow2';
-    my $openqa_url = get_var('OPENQA_URL', autoinst_url);
-    if (get_var('HDD_SWTPM_LEGACY')) {
-        my $hdd_swtpm_legacy = get_required_var('HDD_SWTPM_LEGACY');
-        my $sample_file = 'swtpm/swtpm_legacy.xml';
-        # Since this randomly fails, we retry 10 times each time adding a delay before failing and exiting.
-        my $times = 10;
-        ($times-- && sleep 60) while (script_run("wget -c -P $image_path $openqa_url/assets/hdd/$hdd_swtpm_legacy", 900) != 0 && $times);
-        die "Couldn't download $image_path $openqa_url/assets/hdd/$hdd_swtpm_legacy" unless $times;
-        assert_script_run("mv $image_path/$hdd_swtpm_legacy $image_path/$legacy_image");
-        assert_script_run("wget --quiet " . data_url($sample_file) . " -P $image_path");
-    }
-    elsif (get_var('HDD_SWTPM_UEFI')) {
-        my $hdd_swtpm_uefi = get_required_var('HDD_SWTPM_UEFI');
-        my $sample_file = 'swtpm/swtpm_uefi';
-        $sample_file .= is_aarch64 ? '_aarch64.xml' : '.xml';
-        assert_script_run("wget -c -P $image_path $openqa_url/assets/hdd/$hdd_swtpm_uefi", 900);
-        assert_script_run("mv $image_path/$hdd_swtpm_uefi $image_path/$uefi_image");
-        assert_script_run("wget --quiet " . data_url($sample_file) . " -P $image_path");
+
+    # Download the pre-installed guest image with retries
+    my $attempts = 10;
+    while ($attempts-- > 0) {
```

Comment:
```Perl
while ($attempts--)
```
is enough :)

Reply:
it's not the same behavior though (10 vs 11 runs) :smile:

Reply:
_"Melius est abundare quam deficere"_ (cit.)
both loops stops when `$attempts` is 0 no ?

Reply:
you're right! I thought it'd do also the 0, but it's not the case. FYI https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/21246